scripts/buildtools/check_tests.py

```diff
@@ -0,0 +1,116 @@
+        elif 'output' in testDict["Action"]:
+            if '--- SKIP' in testDict['Output'] and 'due to partitioning' not in AllTestResults[fullTestName]['last_output']:
+                AllTestResults[fullTestName]['skipped_intentionally'] = True
+                if '=== RUN' in AllTestResults[fullTestName]['last_output']:
+                    AllTestResults[fullTestName]['skipped_reason'] = 'No reason given. PLEASE CHECK!!!'
+                else:
+                    AllTestResults[fullTestName]['skipped_reason'] = AllTestResults[fullTestName]['last_output'].strip()
+
+            AllTestResults[fullTestName]['last_output'] = testDict['Output']
+
+            
+
+f.close()
+
+# === Calculate results ===
+
+# Sort and print messages with colored prefix
+RED_TEXT_COLOR = "\033[0;31m"
+GREEN_TEXT_COLOR = "\033[0;32m"
+YELLOW_TEXT_COLOR = "\033[0;33m"
+NORMAL_TEXT_COLOR = "\033[0;0m"
+def printColor(message, color=NORMAL_TEXT_COLOR):
+    print(f"{color}{message}{NORMAL_TEXT_COLOR}")
+
+# Record error message for sys.exit(errorMessage)
+errorMessage = ''
+
+# Check for tests that ran multiple times
+printColor("=========== RAN MULTIPLE TIMES ===================", YELLOW_TEXT_COLOR)
+listOfMultipleRuns = []
+for x in AllTestResults:
+    if AllTestResults[x]['ran'] > 1:
+        listOfMultipleRuns.append(x + " -- ran " + str(AllTestResults[x]['ran']) + " times. (Can probably be fixed by adding \"partitiontest.PartitionTest()\")")
+countMultipleRuns = len(listOfMultipleRuns) 
+if countMultipleRuns:
+    printColor(f"The above {countMultipleRuns} tests ran multiple times:", RED_TEXT_COLOR)
+    [printColor(f"{x}", RED_TEXT_COLOR) for x in sorted(listOfMultipleRuns)]
+    printColor(f"The above {countMultipleRuns} tests ran multiple times:", RED_TEXT_COLOR)
+else:
+    printColor("All tests that ran, ran only once ... OK", GREEN_TEXT_COLOR)
+printColor("==================================================\n", YELLOW_TEXT_COLOR)
+
+# Check intentionally skipped tests
+printColor("============= INTENTIONALLY SKIPPED ==============", YELLOW_TEXT_COLOR)
+listOfSkippedIntentionally = []
+[listOfSkippedIntentionally.append(x) for x in AllTestResults if (AllTestResults[x]['ran'] == 0 and AllTestResults[x]['skipped_intentionally'] == True )]
+countSkippedIntentionally = len(listOfSkippedIntentionally)
+if countSkippedIntentionally:
+    printColor(f"The following {countSkippedIntentionally} tests were skipped intentionally:", YELLOW_TEXT_COLOR)
+    [printColor(f"{x} -- skipped intentionally (please double check) Reason: {AllTestResults[x]['skipped_reason']}", YELLOW_TEXT_COLOR) for x in sorted(listOfSkippedIntentionally)]
+    printColor(f"The above {countSkippedIntentionally} tests were skipped intentionally:", YELLOW_TEXT_COLOR)
+else:
+    printColor("No tests skipped intentionally.", GREEN_TEXT_COLOR)
+printColor("==================================================\n", YELLOW_TEXT_COLOR)
+
+# Check tests unintentionally (due to partition)
```

Comment:
The term "unintentionally" is very confusing. What does it mean to have a test unintentionally skipped? 
If it means a test that should have run but did not run and got skipped, because of a bug in the testing environment, then it should be reported as such: error in the testing environment, with all the visible alarms to get it fixed.

Reply:
Unintentionally means doing it not intentionally, which means not on purpose. It's not necessarily an error in the testing environment, it is most likely a user error, but not always. Dev accidentally put in multiple partitiontest lines for example.

Reply:
i missed a word there, should've been Check tests skipped unintentionally (due to partition)